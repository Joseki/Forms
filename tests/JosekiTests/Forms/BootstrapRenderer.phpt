<?php

/**
 * Test:  Joseki\Forms\BootstrapRenderer.
 */

namespace JosekiTests\Forms;

use Joseki;
use Joseki\Forms\Form;
use Nette;
use Tester;
use Tester\Assert;

require_once __DIR__ . '/../bootstrap.php';

class Translator implements \Nette\Localization\ITranslator
{
    function translate($message, $count = null)
    {
        return strtoupper($message);
    }
}

class BootstrapRendererTest extends Tester\TestCase
{
    public function testUploadTranslate()
    {
        $renderer = new Joseki\Forms\BootstrapRenderer();

        $form = new Form;
        $form->setTranslator(new Translator);
        $form->setRenderer($renderer);
        $form->addUpload('file');
        Assert::matchFile(
            __DIR__ . '/data/BootstrapRenderer.upload1.expected',
            (string)$form
        );

        Joseki\Forms\Messages::$messages[Form::FIND_FILE] = 'aaa';
        Joseki\Forms\Messages::$messages[Form::CHANGE_FILE] = 'bbb';
        Joseki\Forms\Messages::$messages[Form::REMOVE_FILE] = 'ccc';

        Assert::matchFile(
            __DIR__ . '/data/BootstrapRenderer.upload2.expected',
            (string)$form
        );
    }

}

\run(new BootstrapRendererTest());
