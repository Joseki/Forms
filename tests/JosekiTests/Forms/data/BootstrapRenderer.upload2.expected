<form action="" method="post" enctype="multipart/form-data" class="form-horizontal">
<div class="well form-container col-md-push-2 col-md-8">

<div class="form-group">
	<div class="col-sm-4 control-label"><label for="frm-file"></label></div>

	<div class="col-sm-8">
        <div class="fileinput fileinput-new input-group " data-provides="fileinput">
            <div class="form-control" data-trigger="fileinput">
                <span class="fileinput-filename"></span>
            </div>
            <span class="input-group-addon btn btn-default btn-file">
                <span class="fileinput-new">AAA</span>
                <span class="fileinput-exists">BBB</span>
                <input type="file" name="file" id="frm-file">
            </span>
            <a href="#" class="input-group-addon btn btn-default fileinput-exists" data-dismiss="fileinput">CCC</a>
        </div>
    </div>
</div>

</div>
<div><!--[if IE]><input type=IEbug disabled style="display:none"><![endif]--></div>
</form>
